keps/1-kubernetes-enhancement-proposal-process.md

```diff
@@ -221,106 +218,34 @@ Metadata items:
-sufficient value to the community.
+A KEP is has the following states
+
+- `Draft`: The KEP has been proposed and is actively being defined.
+- `Deferred`: The KEP is proposed but not actively being worked on.
+- `Approved`: Using lazy consensus, the approvers have approved this KEP.
+  It will move forward to implementation.
+- `Rejected`: The approvers and authors have decided that this KEP is not moving forward.
+  The KEP is kept around as a historical document.
+- `Withdrawn`: The KEP has been withdrawn by the authors.
+- `Final`: The KEP has been implemented and is no longer actively changed.
+- `Replaced`: The KEP has been replaced by a new KEP.
+  The `superseded-by` metadata value should point to the new KEP.
 
 ### Git and GitHub Implementation
 
-Practically an KEP would be implemented as a pull request to a central repository
-with the following example structure
-
-```
-├── 0000-kep-template.md
-├── CODEOWNERS
-├── index.md
-├── sig-architecture
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── sig-network
-│   ├── deferred
-│   ├── kube-dns
-│   ├── orphaned
-│   └── retired
-├── sig-node
-│   ├── deferred
-│   ├── kubelet
-│   ├── orphaned
-│   └── retired
-├── sig-release
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── sig-storage
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── unsorted-to-be-used-by-newcomers-only
-└── wg-resource-management
-    ├── deferred
-    ├── orphaned
-    └── retired
-```
-
-where each SIG or working group is given a top level directory with subprojects
-maintained by the SIG listed in sub directories. For newcomers to the community
-an `unsorted-to-be-used-by-newcomers-only` directory may be used before an KEP
-can be properly routed to a SIG although hopefully if discussion for a potential
-KEP begins on the mailing lists proper routing information will be provided to
-the KEP author. Additionally a top level index of KEPs may be helpful for people
-looking for a complete list of KEPs. There should be basic CI to ensure that an
-`index.md` remains up to date.
-
-Ideally no work would begin within the repositories of the Kubernetes organization
-before a KEP has been approved by the responsible SIG or working group. While the
-details of how SIGs organize their work is beyond the scope of this proposal one
-possibility would be for each charter SIG to create a top level repository within
-the Kubernetes org where implementation issues managed by that SIG would be filed.
+KEPs are checked into the community repo under the `/kep` directory.
+KEPs can either be put at the top level of that directory or in a SIG specific subdirectory.
+KEPs in SIG specific subdirectories have limited impact outside of the SIG and can leverage SIG specific OWNERS files.
+
+New KEPs can be checked in with a file name in the form of `draft-YYYYMMDD-my-title.md`.
+As significant work is done on the KEP the authors can assign a KEP number.
```

Comment:
do you mean editors can assign a KEP number? Really anyone should be able to submit a PR to assign the KEP number since it's pure administrivia (in my opinion) once the KEP has been approved by the SIG

Reply:
Does a KEP have to be "approved" by a SIG to be created?  I would assume that things get approved once the direction is agreed upon.  Until then the KEP is a draft that is being worked on.

I'm happy with assigning a KEP number sooner rather than later.

Reply:
I think I envisioned that a SIG would signal resolution on {approved, rejected, deferred} by processing the PR
{approved, deferred} -> merge PR
{rejected} -> close PR

Reply:
We had already decided that merging a PR does not equal approval.  We need to have a draft state as a working state before the KEP is "approved".

Reply:
I believe I'm suggesting that merging the PR indicates that the KEP has been considered by the SIG and now lives in one of `{approved, deferred, rejected}`. Making that an invariant might also help us build automated linting of KEPs when we get there. If we find, and here is one place we can pretty easily be data driven, that our PRs are dragging on and the comments are exploding, besides of course the case were a small group of people are generating a fair amount of traffic, we should try tweaking the process then. I'd be happy to work with you to collect that information

Reply:
This is the nut of the disagreement here, IMO.  

My understanding is that we had already decided that "check in early" was the way we were going.  Making "check in early" square with "checked in equal approved" isn't possible.

The template already had "check in early" as part of the instructions at the very top.

I'm happy to continue to discuss this but I'm pretty set on "check in early" being the process here.